mamba/mamba/mamba_shell_init.py

```diff
@@ -81,12 +81,13 @@ def shell_init(args):
         return initialize_dev(shell)
 
     else:
-        for_user = args.user
-        if not (args.install and args.user and args.system):
-            for_user = True
-        # no_user removed in Conda 23.1.0, see https://github.com/mamba-org/mamba/issues/2248
-        if hasattr(args, "no_user") and args.no_user:
-            for_user = False
+        if hasattr(args, "no_user"):
+            # this seems to be conda < 23.1.0
+            # the `no_user` flag, if set, gets erroneously set to False
+            for_user = (args.no_user != False and args.user != False and
+                        not args.system)
```

Comment:
I think I'll replace this with the old code even though it is broken for `--no-user`.  The fix is just so non-intuitive that I think it's maybe better if things are broken for older conda as they used to be.

Reply:
Agreed! Let’s just roll back to the old (broken) code since our „fix“ in #2249 didn’t  actually fix anything